packages/libs/codemodel/.resources/all-in-one/json/code-model.json

```diff
@@ -2191,12 +2191,71 @@
         "authenticationRequired": {
           "description": "indicates that the API surface requires authentication",
           "type": "boolean"
+        },
+        "schemes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SecurityScheme"
+          }
         }
       },
       "defaultProperties": [],
       "additionalProperties": false,
       "required": [
-        "authenticationRequired"
+        "authenticationRequired",
+        "schemes"
+      ]
+    },
+    "SecurityScheme": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "scopes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "headerName": {
+          "type": "string"
+        }
```

Comment:
Scopes we need only for token credential and header name we need only for key credential, so why not move these properties into their respective classes?

Reply:
Do you mean have this instead of an array
```ts
interface SecuritySchemes {
  aadToken?: AADTokenSecurityScheme;
  azureKey?: AzureKeySecurityScheme;
}
```

The array worked great in typescript because you had unions but as can't use them this might be better

Reply:
No, I meant why not move "scopes" property inside `AADTokenSecurityScheme` and "headerName" inside `AzureKeyScheme`?  

Do we need "scopes" for `AzureKeyScheme` and "headerName"  for `AADTokenSecurityScheme`?

Reply:
it is already like that but how do you can't represent the union of those 2 types in csharp so I made the parent type include everything